src/main/java/robosky/ether/mixin/EntityContextImplMixin.java

```diff
@@ -0,0 +1,46 @@
+package robosky.ether.mixin;
+
+import net.minecraft.item.ItemStack;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityContextImpl;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.util.Lazy;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import robosky.ether.block.BlockRegistry;
+import robosky.ether.iface.BossDoorwayContext;
+
+@Mixin(EntityContextImpl.class)
+public abstract class EntityContextImplMixin implements BossDoorwayContext {
+
+    // it is very important that the lazy is only initialized after
+    // item registration. EntityContextImpl is loaded before Uplands registers
+    // its items, so a lazy field is needed.
+    @Unique
+    private static final Lazy<ItemStack> doorway =
```

Comment:
Why does this need to be a field in particular?

Reply:
A new EntityContext is constructed potentially every tick. Constructing a new ItemStack every tick seems wasteful. I thought about using an item tag instead, what are your thoughts on that?